pkg/spec/util.go

```diff
@@ -0,0 +1,214 @@
+package spec
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// This function takes an map as an input which stores the JSON tag as key and
+// the names of the structs holding that tag as the value. The passed in
+// blaclisted tags are ignored and are not populated at all.
+func getUnmarshalJSONTagsMap(inputMap map[string][]string, inputStruct interface{}, blacklistedTags []string) error {
+
+	// Checking if input is pointer to struct
+	if err := isPointerToStruct(inputStruct); err != nil {
+		return errors.Wrap(err, "Input parameter type mismatch")
+	}
+
+	val := reflect.ValueOf(inputStruct).Elem()
+
+	for i := 0; i < val.NumField(); i++ {
+		field := val.Type().Field(i)
+
+		// We do not need to store JSON tags like "inline" or "omitempty",
+		// instead we only need JSON tag which is used for unmarshalling the
+		// YAML. It's safe to assume that, that JSON tag is the one that
+		// is specified first.
+		// Let's split the JSON tag on ",", and store the first element of the
+		// resulting array.
+
+		unmarshalJsonTag := strings.SplitN(field.Tag.Get("json"), ",", 2)[0]
```

Comment:
you can use the library https://github.com/fatih/structtag this helps you parse json tags

I have used it when working on `spec.go` parsing here as:

[`structtag.Parse(j)`](https://github.com/surajssd/kedgeSchema/blob/17adbe3d19a6a995cb957b11bd7aa159a558dd6b/parsego.go#L338)

It basically parses string and gives you list of all all tags, giving an index like this might cause panics later when we encounter the use cases which we are not thinking now!